File: app/src/main/java/com/mxt/anitrend/base/custom/view/text/SingleLineTextView.java
package com.mxt.anitrend.base.custom.view.text;

import android.content.Context;
import android.text.TextUtils;
import android.util.AttributeSet;

import androidx.appcompat.widget.AppCompatTextView;

import com.mxt.anitrend.base.interfaces.view.CustomView;

/**
 * Created by max on 2017/06/24.
 * Single line text view widget
 */

public class SingleLineTextView  extends AppCompatTextView implements CustomView {

    public SingleLineTextView(Context context) {
        super(context);
        onInit();
    }

    public SingleLineTextView(Context context, AttributeSet attrs) {
        super(context, attrs);
        onInit();
    }

    public SingleLineTextView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        onInit();
    }

    /**
     * Clean up any resources that won't be needed
     */
    @Override
    public void onViewRecycled() {

    }

    /**
     * Optionally included when constructing custom views
     */
    @Override
    public void onInit() {
        setSingleLine(true);
        setEllipsize(TextUtils.TruncateAt.END);
    }
}

File: app/src/main/java/com/mxt/anitrend/base/custom/view/text/SingleLineTextView.kt
package com.mxt.anitrend.base.custom.view.text

import android.content.Context
import android.text.TextUtils
import android.util.AttributeSet
import com.google.android.material.textview.MaterialTextView
import com.mxt.anitrend.base.interfaces.view.CustomView

/**
 * Created by max on 2017/06/24.
 * Single line text view widget
 */
open class SingleLineTextView
@JvmOverloads
constructor(
    context: Context,
    attrs: AttributeSet? = null,
    defStyleAttr: Int = 0,
) : MaterialTextView(context, attrs, defStyleAttr),
    CustomView {
    init {
        onInit()
    }

    override fun onViewRecycled() = Unit

    override fun onInit() {
        isSingleLine = true
        ellipsize = TextUtils.TruncateAt.END
    }
}
